include/Operator/SecondQuantized/USQHamiltonian.hpp

```diff
@@ -162,95 +158,101 @@ class USQHamiltonian {
     }
 
     /**
-     *  In-place transform the matrix representations of the alpha components of the unrestricted Hamiltonian
+     *  In-place transform the matrix representations of a single spin component of the unrestricted Hamiltonian
+     * 
+     *  @param component                    the spin component
      *
      *  @param T    the transformation matrix between the old and the new orbital basis
      */
-    void transformAlpha(const TransformationMatrix<Scalar>& T) {
+    void transform(const TransformationMatrix<Scalar>& T, const SpinComponent& component) {
 
-        this->sq_hamiltonian_alpha.transform(T);
+        this->sq_hamiltonians[component].transform(T);
 
         // Transform the mixed two-electron operators and their total
         auto new_two_electron_parameters = this->total_two_op_mixed.parameters();
 
-        // transform the two electron parameters "g_aabb" to "g_a'a'bb"
-        new_two_electron_parameters.template matrixContraction<Scalar>(T, 0);
-        new_two_electron_parameters.template matrixContraction<Scalar>(T, 1);
+        // transform the two electron parameters "g_aabb" to "g_a'a'bb" when alpha is chosen or to "g_aab'b'" for beta.
+        new_two_electron_parameters.template matrixContraction<Scalar>(T, component*2);
+        new_two_electron_parameters.template matrixContraction<Scalar>(T, component*2 + 1);
         this->total_two_op_mixed = ScalarSQTwoElectronOperator<Scalar> ({new_two_electron_parameters});
 
         for (auto& two_op : this->two_op_mixed) {
 
             auto new_two_electron_parameters = two_op.parameters();
             // transform the two electron parameters "g_aabb" to "g_a'a'bb"
-            new_two_electron_parameters.template matrixContraction<Scalar>(T, 0);
-            new_two_electron_parameters.template matrixContraction<Scalar>(T, 1);
+            new_two_electron_parameters.template matrixContraction<Scalar>(T, component*2);
+            new_two_electron_parameters.template matrixContraction<Scalar>(T, component*2 + 1);
             two_op = ScalarSQTwoElectronOperator<Scalar> ({new_two_electron_parameters});
```

Comment:
I really like what you did here!

Improving understandability for newer users further, I would suggest to define

```cpp
first_contraction_index = 2*component;
second_contraction_index = 2*component + 1;
```

to make the meaning clearer.